Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py

```diff
@@ -756,6 +1023,795 @@ def mitigate_threat_command(client: Client, args: dict) -> CommandResults:
+    )
+
+    # Make request and get raw response
+    iocs, pagination = client.get_iocs_request(query_params)
+
+    if pagination['nextCursor'] is not None:
+        demisto.results("Use the below cursor value to get the next page iocs \n {}". format(pagination['nextCursor']))
+
+    if iocs:
+        # Parse response into context & content entries
+        for ioc in iocs:
+            context_entries.append({
+                'UUID': ioc.get('uuid'),
+                'Creator': ioc.get('creator'),
+                'Name': ioc.get('name'),
+                'Value': ioc.get('value'),
+                'Description': ioc.get('description'),
+                'Type': ioc.get('type'),
+                'External Id': ioc.get('externalId'),
+                'Source': ioc.get('source'),
+                'Upload Time': ioc.get('uploadTime'),
+                'Valid Until': ioc.get('validUntil')
+            })
+
+    return CommandResults(
+        readable_output=tableToMarkdown('Sentinel One - Getting List of IOCs', context_entries, removeNull=True,
+                                        metadata='Provides summary information and details for all iocs that matched '
+                                                 'your search criteria.', headerTransform=pascalToSpace),
+        outputs_prefix='SentinelOne.IOC',
+        outputs_key_field='UUID',
+        outputs=context_entries,
+        raw_response=iocs)
+
+
+def create_power_query(client: Client, args: dict) -> CommandResults:
+    """
+    Create the power query and get the events or get the query ID. Relavent for API version 2.1
+    """
+    context_entries = []
+
+    # Get arguments
+    limit = int(args.get('limit', 1000))
+    query = args.get('query')
+    from_date = args.get('from_date')
+    to_date = args.get('to_date')
+
+    # Make request and get raw response
+    response = client.create_power_query_request(limit, query, from_date, to_date)
+
+    if response['status'] == 'RUNNING':
+        context_entries.append({
+            'queryId': response['queryId']
+        })
+        meta = "Ping a Deep Visibility Power Query using the queryId"
+
+    elif response['status'] == 'FINISHED':
+        for row in response['data']:
+            temp = {}
+            for i in range(len(row)):
+                temp.update({response['columns'][i]['name']: row[i]})
+            context_entries.append(temp)
+
+        meta = 'Provides summary information and details aboput the power query and its id \n your search criteria.'
+
+    return CommandResults(
+        readable_output=tableToMarkdown('Sentinel One - Create a Power Query and Get QueryId', context_entries, removeNull=True,
+                                        metadata=meta, headerTransform=pascalToSpace),
+        outputs_prefix='SentinelOne.PowerQuery',
+        outputs=context_entries,
+        raw_response=response)
+
+
+def ping_power_query(client: Client, args: dict) -> CommandResults:
```

Comment:
we can't poll until the data is not none, the reason is it will never return the data object which is not None once the data is not available means it is never available, this will result in an infinite loop